pageserver/src/tenant_mgr.rs

```diff
@@ -252,21 +250,73 @@ fn create_repo(
     wal_redo_manager: Arc<dyn WalRedoManager + Send + Sync>,
     remote_index: RemoteIndex,
 ) -> anyhow::Result<Arc<Repository>> {
-    let repo_dir = conf.tenant_path(&tenant_id);
-    ensure!(
-        !repo_dir.exists(),
-        "cannot create new tenant repo: '{}' directory already exists",
-        tenant_id
+    let target_tenant_directory = conf.tenant_path(&tenant_id);
+    anyhow::ensure!(
+        !target_tenant_directory.exists(),
+        "cannot create new tenant repo: '{tenant_id}' directory already exists",
+    );
+
+    let temporary_tenant_dir =
+        path_with_suffix_extension(&target_tenant_directory, TEMP_FILE_SUFFIX);
+    debug!(
+        "Creating temporary directory structure in {}",
+        temporary_tenant_dir.display()
     );
 
+    let temporary_tenant_timelines_dir = rebase_directory(
+        &conf.timelines_path(&tenant_id),
+        &target_tenant_directory,
+        &temporary_tenant_dir,
+    )?;
+    let temporary_tenant_config_path = rebase_directory(
+        &TenantConf::path(conf, tenant_id),
+        &target_tenant_directory,
+        &temporary_tenant_dir,
+    )?;
+
     // top-level dir may exist if we are creating it through CLI
-    crashsafe_dir::create_dir_all(&repo_dir)
-        .with_context(|| format!("could not create directory {}", repo_dir.display()))?;
-    crashsafe_dir::create_dir(conf.timelines_path(&tenant_id))?;
-    info!("created directory structure in {}", repo_dir.display());
+    crashsafe_dir::create_dir_all(&temporary_tenant_dir).with_context(|| {
+        format!(
+            "could not create temporary tenant directory {}",
+            temporary_tenant_dir.display()
+        )
+    })?;
+    // first, create a config in the top-level temp directory, fsync the file
+    Repository::persist_tenant_config(&temporary_tenant_config_path, tenant_conf)?;
```

Comment:
IMO makes sense for `persist_tenant_config` to sync the file too. Maybe even with the parent. Probably can work the same way as it works with [save_metadata](https://github.com/neondatabase/neon/blob/main/pageserver/src/layered_repository/metadata.rs#L186) with `first_save` argument that triggers fsync on parent dir